packages/ra-ui-materialui/src/button/SelectAllButton.spec.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Basic, Label, Limit } from './SelectAllButton.stories';
+
+describe('<SelectAllButton />', () => {
+    it('should render a "Select All" button', async () => {
+        render(<Basic />);
+        await screen.findByText('War and Peace');
+        fireEvent.click(screen.getAllByRole('checkbox')[0]);
+        await screen.findByRole('button', { name: 'Select all' });
+    });
+
+    it('should not render a "Select All" button if not all checkboxes are checked', async () => {
+        render(<Basic />);
+        await screen.findByText('War and Peace');
+        fireEvent.click(screen.getAllByRole('checkbox')[0]);
+        await screen.findByText('10 items selected');
+        expect(
+            screen.queryByRole('button', { name: 'Select all' })
+        ).not.toBeNull();
```

Comment:
You should use the same assertion as the previous test (`await screen.findByRole('button', { name: 'Select all' });`) to make them more easily comparable.